core/controllers/voice_artist_test.py

```diff
@@ -416,3 +418,188 @@ def test_cannot_deassign_random_user_from_voice_artist(self):
+
+        with utils.open_file(
+                os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_MP3),
+                'rb', encoding=None
+        ) as f:
+            raw_audio = f.read()
+        response_dict = self.post_json(
+            '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+            {'filename': supplied_filename},
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'],
+            'Invalid filename extension: it should have '
+            'one of the following extensions: %s'
+            % list(feconf.ACCEPTED_AUDIO_EXTENSIONS.keys()))
+
+    def test_mutagen_bad_audio(self):
+        response_dict = self.post_json(
+            '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+            {'filename': 'test.mp3'},
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(
+                ('raw_audio_file', 'unused_filename', 'non_audio_data'),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertEqual(
+            response_dict['error'], 'Audio not recognized as a mp3 file')
+
+    def test_detect_non_audio(self):
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            'rb', encoding=None
+        ) as f:
+            raw_audio = f.read()
+
+        with self.accepted_audio_extensions_swap:
+            response_dict = self.post_json(
+                '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+                {'filename': self.TEST_AUDIO_FILE_FLAC},
+                csrf_token=self.csrf_token,
+                expected_status_int=400,
+                upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+            )
+        self.assertEqual(
+            response_dict['error'], 'Audio not recognized as a flac file')
+
+    def test_too_long(self):
+        with utils.open_file(
+            os.path.join(
+                feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_OVER_MAX_LENGTH),
+            'rb', encoding=None
+        ) as f:
+            raw_audio = f.read()
+        response_dict = self.post_json(
+            '%s/%s' % (self.AUDIO_UPLOAD_URL_PREFIX, self.EXP_ID),
+            {'filename': 'test.mp3'},
+            csrf_token=self.csrf_token,
+            expected_status_int=400,
+            upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
+        )
+        self.assertEqual(response_dict['status_code'], 400)
+        self.assertIn(
+            'Audio files must be under %s seconds in length'
+            % feconf.MAX_AUDIO_FILE_LENGTH_SEC, response_dict['error'])
+
+    def test_extensions_do_not_match(self):
+        mismatched_filename = 'test.flac'
```

Comment:
Perhaps `incorrect_filename` might be better here.

Also, it seems like the filenames **and** the extensions are different in this test. Should we rename the description as such and have another test that checks if the same fails if the filename is the same but extension is different? And one to check the scenario where the extensions are the same but the filename is different.